adapters/rubicon/rubicon.go

```diff
@@ -473,3 +503,174 @@ func NewRubiconAdapter(config *adapters.HTTPAdapterConfig, uri string, xuser str
 		XAPIPassword: xpass,
 	}
 }
+
+func (a *RubiconAdapter) MakeRequests(request *openrtb.BidRequest) ([]*adapters.RequestData, []error) {
+	numRequests := len(request.Imp)
+	errs := make([]error, 0, len(request.Imp))
+	var err error
+
+	requestData := make([]*adapters.RequestData, 0, numRequests)
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("User-Agent", "prebid-server/1.0")
+
+	for i := 0; i < numRequests; i++ {
+		request.Imp = request.Imp[i : i+1]
+
+		var bidderExt adapters.ExtImpBidder
+		if err = json.Unmarshal(request.Imp[0].Ext, &bidderExt); err != nil {
+			errs = append(errs, err)
+			return nil, errs
+		}
+
+		var rubiconExt openrtb_ext.ExtImpRubicon
+		if err = json.Unmarshal(bidderExt.Bidder, &rubiconExt); err != nil {
+			errs = append(errs, err)
+			return nil, errs
+		}
+
+		if rubiconExt.AccountId == 0 {
```

Comment:
I struggled with an issue here which you all might want to weigh in on.

All OpenRTB requests are filtered by [your JSON schema](https://github.com/prebid/prebid-server/blob/master/static/bidder-params/rubicon.json). If some publisher sends a request which doesn't match, then the endpoint returns a 400, and your bidder will never be called.

Given that, do you think the Bidder code also needs to run these checks? I went back and forth on it, and I lean towards "no", although I can see some benefits to it as well.

Did you have any thoughts here?

Reply:
the json schema doesn't seem to rule our zeroes for ints. unless there's a way to add that check in the schema we need to check for zeroes so we don't waste time sending an invalid request.

Reply:
Are negative values allowed? If not, you can set the min value to 1: https://spacetelescope.github.io/understanding-json-schema/reference/numeric.html#range